domain/info.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/ddl"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/printer"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	// ServerInformation store server information such as IP, port and so on.
+	ServerInformation = "/tidb/server/Info"
+)
+
+// infoSyncer stores server Info to PD when server start and delete when server down.
+type infoSyncer struct {
+	etcdCli        *clientv3.Client
+	info           *ServerInfo
+	serverInfoPath string
+}
+
+// ServerInfo is server static information.
+// It will not update when server running. So please only put static information in ServerInfo struct.
+type ServerInfo struct {
+	ServerVersionInfo
+	ID         string `json:"ddl_id"`
+	IP         string `json:"ip"`
```

Comment:
Miss listening port.

Reply:
`StatusPort` is 10080.
listening port you mean 4000?

Reply:
Yes, 4000 is the default one.